### DISCUSSION TITLE
Import check-in data from external source

### DISCUSSION THREAD

**Discussion by raphaelm** [pretix/pretix#1962] [Ideas]

**Comment by jfwiebe** [comment]
Did I get right that the checkin-data is currently only stored if the pretixdroid plugin is enabled? wouldn't it be nice to move the "checked in" flag to the core so that it can be interacted with it in various ways?

**Comment by raphaelm** [comment]
Heh, I already did that some time ago in fa0bd5e89e2fe1e3fb546fe2a3ec1856c6c4e2e6 :)

Also, we did this import that is mentioned here from [c6sh](https://github.com/c3cashdesk/c6sh) data using a very simple management command, so the remaining scope of this issue would be to wrap this in some small user interface (could be a plugin, too).

```python
import json

from django.core.management.base import BaseCommand
from pretix.base.models import Event, OrderPosition, Checkin
import dateutil.parser


class Command(BaseCommand):
    help = "Import checkins from c6sh"

    def add_arguments(self, parser):
        parser.add_argument('event_id', type=int)
        parser.add_argument('filename', type=str)

    def handle(self, *args, **options):
        event = Event.objects.get(pk=options.get('event_id'))
        with open(options.get('filename'), 'r') as f:
            data = json.load(f)
        for d in data:
            op = OrderPosition.objects.get(secret=d['secret'], order__event=event)
            Checkin.objects.get_or_create(position=op, datetime=dateutil.parser.parse(d['datetime']))
```
